index.js

```diff
@@ -27,24 +29,55 @@ try {
  * @method resolvePackagePathSync
  * @param {string} name name of the dependency module.
  * @param {string} basedir root dir to run the resolve from
- * @param {Boolean} shouldCache (optional) intended to bypass cache defaulting to true
+ * @param {Boolean|CustomCache} (optional)
+ *  * if true: will choose the default global cache
+ *  * if false: will not cache
+ *  * if undefined or omitted, will choose the default global cache
+ *  * otherwise we assume the argument is an external cache of the form:
+ *      cache = {
+ *        PATH: new Map(),
+ *        PATH: new Map(),
+ *        REAL_DIRECTORY_PATH: new Map()
+ *      };
+ *
  * @return {string|null} a full path to the resolved package.json if found or null if not
  */
-module.exports = function resolvePackagePathSync(target, basedir, shouldCache = true) {
-  let pkgPath;
-  target = `${target}/package.json`;
-  let key = `${target}\x00${basedir}`;
+module.exports = function resolvePackagePathSync(target, basedir, _cache) {
```

Comment:
Is there a reason this name ends in 'Sync', other than that 'fs' methods have both? Is there a non-sync version?